### PR TITLE
Update bug fixes

### LINKED ISSUE #176: [PE-D] Removing students feature behavior while in module view
The data used is the sample data of the product.

We are in the CS2103T module now.

According to the user guide, the behavior when I input the command `rm 2 m/CS2103T c/T07` be that the second entry be removed successfully? 2 is the input of the "displayed" list.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: IceWizard4902/ped#8

### LINKED ISSUE #206: ModuleSet not interacting with mv and rm
1. Unable to move a person to another class, if he already has 1 class in the module.
image

2. Removing all the classes from a person's module leaves the data in this state. 
image

### PR DESCRIPTION
**Pull request by chaiwanlin**
closes  #176, #156

### CODE REVIEW

File: src/main/java/seedu/edrecord/storage/JsonAdaptedPerson.java [hunk 1]
```diff
@@ -116,42 +116,45 @@ public Person toModelType() throws IllegalValueException {
...
+                        throw new IllegalValueException(Group.MESSAGE_CONSTRAINTS);
+                    }
+                    if (!modelModule.getGroupSystem().hasGroup(group)) {
+                        throw new IllegalValueException(Group.MESSAGE_DOES_NOT_EXIST);
+                    }
+                    final Group modelGroup = modelModule.getGroup(group);
+                    if (!modelModule.getGroupSystem().hasGroup(modelGroup)) {
+                        modelModule.getGroupSystem().addGroup(modelGroup);
+                    }
+                    moduleSet.addToSet(modelModule);
```
Comment (FergusMok) [inline src/main/java/seedu/edrecord/storage/JsonAdaptedPerson.java:153]:
```suggestion
                    moduleSet.add(model, new Group(group));
```
Also, do not add the model system's module. The model module may have more groups than an individual person

Comment (FergusMok) [reply src/main/java/seedu/edrecord/storage/JsonAdaptedPerson.java:153]: LGTM otherwise

**Review by FergusMok — APPROVED** [review]
LGTM

**Review by marcusc55 — APPROVED** [review]
LGTM

**Comment by caipng** [comment]
@chaiwanlin This PR should be good to merge, just need you to fix the merge conflict.

**Pull request merged**